internal/pkg/db/mongo/data.go

```diff
@@ -52,6 +52,15 @@ func (mc MongoClient) AddEvent(e correlation.Event) (string, error) {
 	s := mc.getSessionCopy()
 	defer s.Close()
 
+	if e.Created == 0 {
+		e.Created = db.MakeTimestamp()
+		e.Modified = e.Created
+	}
+
+	if e.Modified == 0 {
+		e.Modified = db.MakeTimestamp()
+	}
```

Comment:
Should we be setting the `modified` value or leaving it to whatever the caller has specified? I only ask because the Redis implementation does not seem to be doing this, and it would be nice if this is consistent. If we do want to keep this logic, then I think we may need to update this. With this implementation we will overwrite the `modified` value if the `created` value is not set, regardless if the `modified` value has been provided. I think line `57` could be removed. and line `61` could simply be:
```go
e.modified = e.created
```

that way we guarantee that the `created` and `modified` times are the same in the case where neither of the values are provided and are generated by the system. I think that would make more sense to a user who is looking at the data rather than possibly having two different times.

Reply:
Yes.it's great.